users/views.py

```diff
@@ -105,15 +105,19 @@ def get_success_url(self) -> str:
         return reverse("custom-users-list")
 
     def form_valid(self, form: CustomUserCreationForm) -> HttpResponse:
-        user_random_password = \
-            generate_random_string_from_letters_and_digits(UserConstants.RANDOM_PASSWORD_LENGTH.value)
-        user = form.save(user_random_password)
-        user.set_password(user_random_password)
-        user.full_clean()
-        user.save()
-        self._send_create_user_email(user, user_random_password)
-        logger.info(f"New user with id: {user.pk} has been created")
-        messages.success(self.request, ConfirmationMessages.SUCCESSFUL_CREATE_USER_MESSAGE)
+        if not self.request.is_ajax():
+            user_random_password = generate_random_string_from_letters_and_digits(
+                UserConstants.RANDOM_PASSWORD_LENGTH.value
+            )
+            user = form.save()
+            user.set_password(user_random_password)
+            user.is_active = True
+            user.need_change_password = True
+            user.full_clean()
+            user.save()
+            self._send_create_user_email(user, user_random_password)
+            logger.info(f"New user with id: {user.pk} has been created")
+            messages.success(self.request, ConfirmationMessages.SUCCESSFUL_CREATE_USER_MESSAGE)
```

Comment:
Please squash this changes in final version to have only one commit where you are implementing `fom_valid` in views and changing forms. This is similar situation to separate commit with changes in models. Please squash changes that refers to other from the same branch for each situation.

Reply:
Done.